### PR TITLE
feat: add route to create a note

### PR DESCRIPTION
**Pull request by cballevre**
```
### ✨ Features

* add route to create a note

```

This addition is linked to the addition of a FAB button on cozy-home (cc: https://github.com/cozy/cozy-home/pull/1860).

### CODE REVIEW

**Review by zatteo — APPROVED** [review]

File: src/components/notes/new-route.jsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        pathname: `/`
+      })
+      window.location.href = await generateReturnUrlToNotesIndex(
+        client,
+        doc,
+        returnUrl
+      )
+    }
+    createNote()
+  })
```
Comment (Crash--) [inline src/components/notes/new-route.jsx:29]: useEffect without an dep array? 🤔

File: src/components/notes/new-route.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  useEffect(() => {
+    const createNote = async () => {
+      const { data: doc } = await createNoteDocument(client)
+      const returnUrl = generateWebLink({
+        slug: 'notes',
+        cozyUrl: client.getStackClient().uri,
+        subDomainType: isFlatDomain ? 'flat' : 'nested',
+        pathname: `/`
+      })
+      window.location.href = await generateReturnUrlToNotesIndex(
```
Comment (Crash--) [inline src/components/notes/new-route.jsx:22]: Hmm, I think we should not directly change the href. 

Let's do the same we it's already done in Drive: https://github.com/cozy/cozy-drive/blob/953e2c4e25e0c8461b9813f18599d9aa39d38bd2/src/drive/web/modules/drive/Toolbar/components/CreateNoteItem.jsx#L75-L85

File: src/components/notes/new-route.jsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  const { capabilities } = useCapabilities(client)
+  const isFlatDomain = get(capabilities, 'flat_subdomains')
+
+  useEffect(() => {
+    const createNote = async () => {
+      const { data: doc } = await createNoteDocument(client)
+      const returnUrl = generateWebLink({
+        slug: 'notes',
+        cozyUrl: client.getStackClient().uri,
+        subDomainType: isFlatDomain ? 'flat' : 'nested',
```
Comment (Crash--) [inline src/components/notes/new-route.jsx:19]: `useCapabilities` is async. So you need to wait the value before calling the createNote() method. If not, isFlatDomain will always be undefined

File: src/lib/utils.js [hunk 4]
```diff
@@ -30,11 +30,11 @@ export function getReturnUrl() {
  * @param {object} doc - couchdb document for the note/file
  * @return {string} URL where one can edit the note
  */
-export async function generateReturnUrlToNotesIndex(client, doc) {
+export async function generateReturnUrlToNotesIndex(client, doc, returnUrl) {
```
Comment (Crash--) [inline src/lib/utils.js:33]: you should update the jsdoc 🙏

**Review by Crash-- — APPROVED** [review]
Looks better :)

**Pull request merged**